packages/vhd-lib/merge.js

```diff
@@ -267,6 +271,9 @@ class Merger {
         this.#logInfo(`the VHD parent was already renamed`, { parent, mergeTargetChild })
       }
     }
+    if (oldTarget !== undefined) {
+      ;(await VhdAbstract.unlink(handler, oldTarget)).catch(warn)
+    }
```

Comment:
Incorrect in case the parent was already renamed, should be done only when the `rename` succeeds.